src/shared_modules/content_manager/src/components/executionContext.hpp

```diff
@@ -46,18 +47,39 @@ class ExecutionContext final : public AbstractHandler<std::shared_ptr<UpdaterBas
         unsigned int databaseOffset;
         try
         {
-            databaseOffset = std::stoi(context.spRocksDB->getLastKeyValue().second.ToString());
+            databaseOffset =
+                std::stoi(context.spRocksDB->getLastKeyValue(Components::Columns::CURRENT_OFFSET).second.ToString());
         }
         catch (const std::runtime_error&)
         {
             // First execution. Set offset to zero.
             databaseOffset = 0;
-            context.spRocksDB->put(Utils::getCompactTimestamp(std::time(nullptr)), "0");
+            context.spRocksDB->put(
+                Utils::getCompactTimestamp(std::time(nullptr)), "0", Components::Columns::CURRENT_OFFSET);
         }
 
         return databaseOffset;
     }
 
+    /**
+     * @brief Reads and returns the file hash from the database.
+     *
+     * @param context Updater context configured with the database driver.
+     * @return unsigned int Last file hash from the database.
```

Comment:
Typo?
The function returns a string, not an int.

Also, it could mention the special case for an empty string.

Reply:
Yes. Fixed!